lib/redis_cluster/client.rb

```diff
@@ -230,6 +230,33 @@ def refresh_startup_nodes
       @hosts.uniq!
     end
 
+    # Retries an operation @retry_count times for intermittent connection
+    # errors. After exhausting retries, the error that was received on the last
+    # attempt is raised to the user.
+    def retry_intermittent_loop
+      last_error = nil
+
+      for attempt in 0..(@retry_count) do
+        begin
+          yield(attempt)
+
+          # Fall through on any success.
+          return
+        rescue Errno::EACCES, Redis::TimeoutError, Redis::CannotConnectError => e
```

Comment:
I took `Errno::ECONNREFUSED` out of the list because it's rolled up in `Redis::CannotConnectError`.